src/util/signature_utils.js

```diff
@@ -343,7 +345,35 @@ function authorize_client_request(req) {
     req.headers.Authorization = v4.authorization(req.credentials, req.headers['X-Amz-Date']);
 }
 
+// mutual code of S#, STS and IAM services
+// sdk - this can be: object_sdk, sts_sdk or account_sdk
+function authenticate_request_by_service(req, sdk) {
+    const auth_token = make_auth_token_from_request(req);
+    if (auth_token) {
+        auth_token.client_ip = http_utils.parse_client_ip(req);
+    }
+    if (req.session_token) {
+        auth_token.access_key = req.session_token.assumed_role_access_key;
+        auth_token.temp_access_key = req.session_token.access_key;
+        auth_token.temp_secret_key = req.session_token.secret_key;
+    }
+    sdk.set_auth_token(auth_token);
+    check_request_expiry(req);
+}
+
+// mutual code of S#, STS and IAM services
+function authorize_request_account_by_token(token, requesting_account, is_anonymous_requests_are_allowed = false) {
```

Comment:
rename and no default -
```suggestion
function authorize_request_account_by_token(token, requesting_account, is_secret_optional) {
```